main/main.c

```diff
@@ -15,42 +13,328 @@
+#define SYS_CLOCK_10MS (1 / portTICK_PERIOD_MS)
 
-static void initialize_sensors(void)
+#define LORA_INIT_MAX 5				// Num attempts to initialize LoRa
+
+#define xNODE_BME280_FAIL	0x80	// Set when BME280 was not initialized
+#define xNODE_LORA_FAIL		0x10	// Set when LoRa init fails
+#define xNODE_LOW_BATT		0x08	// Set when low battery detected
+#define xNODE_MQTT_FAIL		0x04	// Set when MQTT fails
+
+#define SCHEDULE_SZ 3
+#define PERIOD_BME280 1
+#define PERIOD_BASE PERIOD_BME280			// Set a base for period.
+#define PERIOD_STATUS (PERIOD_BME280 * 10) 	// All other periods are multiple of BASE
+#define PERIOD_SMS (PERIOD_BME280 * 40)		// All other periods are multiple of BASE
+
+#define MAX_DATA_TYPES 33
+
+#if CONFIG_LORA_RECEIVER
+#define MSG_BUFFER_SIZE 128
+#endif
+
+// ------------------------------------+
+//				N V M                  |
+// -----------+---------------+--------+
+// NAMESPACE  |  KEY          | VALUE  |
+// -----------+---------------+--------+
+// radio_link | msg_id        | 8-bit  |
+// radio_link | msg_count     | 8-bit  |
+// radio_link | comms_err_cnt | 8-bit  |
+// radio_link | lowpwr_counter| 16-bit | // TBD: need to update this namespace
+// -----------+---------------+--------+
+
+typedef struct nvs_map
 {
-    esp_err_t bme_rc = ESP_OK;
-    if (ESP_OK != bme280_init_driver(CONFIG_BME280_I2C_ADDRESS))
+	const char *namespace;
+	const char *key0;			// TBD: This needs to be optimized (loop-through)
+	const char *key1;
+	const char *key2;
+	const char *key3;
+} nvs_map_t;
+
+typedef struct bme280
+{
+	double temp_raw;
+	double press_raw;
+	double hum_raw;
+	uint16_t comms_err_cnt;					// Communication error counter
+} bme280_t;
+
+typedef struct bq27441
+{
+	uint16_t voltage;
+} bq27441_t;
+
+typedef struct sensor
+{
+	bme280_t bme280;
+	bq27441_t bq27441;
+} sensor_t;
+
+// dummy struct
+typedef struct radio
+{
+	uint8_t msg_counter;
+	uint8_t msg_id;
+	uint16_t comms_err_cnt;	// Communication error counter
+	QueueHandle_t queue_sch;
+	QueueHandle_t queue_rx;
+} radio_t;
+
+typedef struct system
+{
+	QueueHandle_t queue;
+	uint16_t sleep_time;
+	uint16_t life_cycle;
+} system_t;
+
+typedef struct node
+{
+	uint8_t status;  //TBD may need to config of 'node' and status go to system
+	system_t system;
+	sensor_t sensor;
+	radio_t radio;
+} node_t;
+
+typedef struct buzzverse_link
+{
+	void (*handler)(void);
+	void (*sender)(void);
+} buzzverse_link_t;
+
+typedef struct schedule
+{
+		DataType type;
+		uint32_t period;
+} schedule_t;
+
+typedef enum sys_msg
+{
+	SYS_RUN_LOWPWR_MODE = 0,
+	SYS_LORA_FAIL = 1
+} sys_msg_t;
+
+typedef enum sch_msg
+{
+	SCH_START = 0,
+	SCH_RADIO_POWER_OFF = 1
+} sch_msg_t;
+
+/*
+ * Local static vars, functions declarations
+ */
+static void task_radio_scheduler(void *pvParameters);
+static nvs_handle_t sys_mount_nvs(const char *namespace);
+static void sys_umount_nvs(nvs_handle_t handle);
+
+/* BuzzVerse link handlers/senders */
+static void handler_unknown(void);

```

Comment:
but if the underlying structure/array already has buzzverse_link then it will expand to:
buzzverse_link[scheduler[i].type].bv_link_sender_unknown
vs
buzzverse_link[scheduler[i].type].sender_unknown

For now I'm ok for both